pkg/tfbridge/schema.go

```diff
@@ -491,25 +500,45 @@ func MakeTerraformConfigFromInputs(inputs map[string]interface{}) (*terraform.Re
 // MakeTerraformAttributes converts a Pulumi property bag into its Terraform equivalent.  This requires
 // flattening everything and serializing individual properties as strings.  This is a little awkward, but it's how
 // Terraform represents resource properties (schemas are simply sugar on top).
-func MakeTerraformAttributes(res *PulumiResource, m resource.PropertyMap,
-	tfs map[string]*schema.Schema, ps map[string]*SchemaInfo, defaults bool) (map[string]string, error) {
+func MakeTerraformAttributes(res *schema.Resource, m resource.PropertyMap, tfs map[string]*schema.Schema,
+	ps map[string]*SchemaInfo, defaults bool) (map[string]string, map[string]interface{}, error) {
+
+	// Strip out any metadata from the inputs.
+	var meta map[string]interface{}
+	if metaProperty, hasMeta := m[metaKey]; hasMeta && metaProperty.IsObject() {
+		metaValue, err := MakeTerraformInputs(nil, nil, metaProperty.ObjectValue(), nil, nil, nil, false, true)
+		if err != nil {
+			return nil, nil, err
+		}
+		meta = metaValue
+	} else if res.SchemaVersion > 0 {
+		// If there was no metadata in the input and this resource has a non-zero schema version, return a meta bag
+		// with the current schema version. This helps avoid migration issues.
+		meta = map[string]interface{}{"schema_version": strconv.Itoa(res.SchemaVersion)}
+	}
+
 	// Turn the resource properties into a map.  For the most part, this is a straight Mappable, but we use MapReplace
 	// because we use float64s and Terraform uses ints, to represent numbers.
-	inputs, err := MakeTerraformInputs(res, nil, m, tfs, ps, nil, defaults, false)
+	inputs, err := MakeTerraformInputs(nil, nil, m, tfs, ps, nil, defaults, false)
```

Comment:
Why is this now passing `nil`?

Reply:
The input this used to pass-`res`--was always `nil` prior to this change, and is now of a different type.